R/sign-commits.R

```diff
@@ -90,3 +105,44 @@ gh_store_key <- function(key, .token = NULL) {
     )
   }
 }
+
+extract_git_option <- function(name) {
+  git_config <- git2r::config()
+  if (is.null(git_config[["local"]][[name]])) {
+    git_config[["global"]][[name]]
+  } else {
+    git_config[["local"]][[name]]
+  }
+}
+
+get_key_candidates <- function(user_name, user_email) {
+  existing_keys <- gpg::gpg_list_keys()
+  if (is.null(user_name) | is.null(user_email)) {
+    subset(
+      existing_keys,
+      existing_keys$name == user_name | existing_keys$email == user_email
```

Comment:
I think this logic needs to be modified, it will always return an empty data frame if either the name or email is `NULL` because `existing_keys$name == NULL` returns `logical(0)`, not a vector of `TRUE`/`FALSE`.

Reply:
thanks, totally missed that! I added some tests for this bug fix and created a separate function so it is easy to write tests.

Reply:
I see the new function in the tests but not in the code itself.

Reply:
really sorry about that, forgot to commit!